game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -134,8 +134,53 @@ def skynet_nodes(self, game: Game) -> list[SkynetNode]:
             skynet_nodes.append(skynet_node)
         return skynet_nodes
 
+    def _update_iads_comms_and_power(
+        self, tgo: TheaterGroundObject, events: GameUpdateEvents
+    ) -> None:
+        assert self.advanced_iads, "_update_iads_comms_and_power requires advanced IADS"
+        is_comm_or_power = IadsRole.for_category(tgo.category).is_comms_or_power
+        assert is_comm_or_power, "Invalid TGO was given for _update_iads_building"
+
+        """ 
+        Delete/Update connections to the comm tower/power station
+        If this function is called, it should imply only 2 possibilities (unless I missed one):
+            1) A capture event occurred, thus the building now belongs to the capturing team
+                => All connections to this building are to the enemy, thus delete them
+                 (mind that no new connections could have been formed since all TGOs were depopulated)
+            2) The building was destroyed during a mission
+                => In this case we don't need to delete the connections
+                    instead we just update the nodes that connect to this building
+                    because those connections are still coming from friendly TGOs
+        Given the above, we should be able to use the following implications:
+            If the building is friendly compared to the node that we're checking
+                => Building was destroyed during mission
+                => Update nodes that connect to this building
+            Otherwise if the building is not friendly compared to the node we're checking
+                => Capture event occurred
+                => Delete all connections to this building
+
+        TODO: clean up the code below by wrapping comm towers and power stations
+                preferably in a class like IadsNetworkNode, keeping a reference to connected nodes
+        """
+        for node in self.nodes:
+            to_delete = []
+            for cID in node.connections:
+                group = node.connections[cID]
+                if group.ground_object is tgo:
+                    if self._is_friendly(node, tgo):
+                        events.update_iads_node(node)
+                    else:
+                        to_delete.append(cID)
+                        events.delete_iads_connection(cID)
+            for cID in to_delete:
+                del node.connections[cID]
+        if not self.iads_config:
+            self._update_network(tgo, events)
+
```

Comment:
@RndName Comments should clear up the question you have in #2281.

Furthermore, line 177 has the same idea behind it as in #2281, i.e. only make advanced connections from nodes to the comm tower or power station in question if we've initialized by range to start with.

The TODO is referring to what I described in #2281 as the hybrid solution, because as you suggested in your first solution, those nearby TGOs to which we could connect won't change after initialization in the sense that there's a maximum number of possible connections (limited by range) for each comm tower or power station. If we keep track of those nodes, updating the network won't require looping over all nodes in the network.